tycho-indexer/src/extractor/dynamic_contract_indexer/euler/metadata_generator.rs

```diff
@@ -0,0 +1,258 @@
+use tycho_common::models::{blockchain::Block, protocol::ProtocolComponent, TxHash};
+
+use crate::extractor::dynamic_contract_indexer::component_metadata::{
+    MetadataError, MetadataRequest, MetadataRequestGenerator, MetadataRequestType, RpcTransport,
+};
+
+pub struct EulerMetadataGenerator {
+    rpc_url: String,
+}
+
+impl EulerMetadataGenerator {
+    pub fn new(rpc_url: String) -> Self {
+        Self { rpc_url }
+    }
+}
+
+impl MetadataRequestGenerator for EulerMetadataGenerator {
+    fn generate_requests(
+        &self,
+        component: &ProtocolComponent,
+        block: &Block,
+    ) -> Result<Vec<MetadataRequest>, MetadataError> {
+        let mut requests = vec![];
+
+        let balance_transport = RpcTransport::new(
+            self.rpc_url.clone(),
+            "eth_call".to_string(),
+            vec![serde_json::json!([
+                  {
+                    "data": "0x0902f1ac", // getReserves()
+                    "to": component.id
+                  },
+                  block.number.to_string() //TODO: should it be formatted as a hex string?
+                ]
```

Comment:
Yes, the pattern is hex string for block number: https://www.quicknode.com/docs/ethereum/eth_getStorageAt